### PR TITLE
ci: Added codespell

### PR DESCRIPTION
**Pull request by marksweb**
## Description

Adds spell checking to pre-commit and github actions.

## Related resources

* #7287

## Checklist

* [x] I have opened this pull request against ``develop``
* [ ] I have added or modified the tests when changing logic
* [x] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [x] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

File: .github/workflows/spelling.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  cancel-in-progress: true
+
+jobs:
+  codespell:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v3
```
Comment (cclauss) [inline .github/workflows/spelling.yml:16]:
```suggestion
        uses: actions/setup-python@v4
```

**Review by fsbraun — APPROVED** [review]

File: .github/workflows/spelling.yml (hunk 1 above)
Comment (marksweb) [reply .github/workflows/spelling.yml]: Thanks @cclauss 👍

File: .github/workflows/spelling.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+          cache: 'pip'
+      - run: python -m pip install codespell
```
Comment (cclauss) [inline .github/workflows/spelling.yml:20]: The codespell action runs in a Docker container which makes these lines unnecessary.
```suggestion
```

Comment (marksweb) [reply .github/workflows/spelling.yml]: Thanks @cclauss 👍

**Pull request merged**